### DISCUSSION TITLE
monorepo: find file in cwd and root not working

### DISCUSSION THREAD

**Discussion by codingluke** [LazyVim/LazyVim#1783] [Q&A] [answered]
I am working on a bigger monorepo where we have multiple js modules in subfolders. Since newly lazyvim is directly switching the root to the submodule, therefore the `<leader>ff` and `<leader>fF` produce the same result. How is it possible to switch the root detection to only detect it by `.git` and not by the root of all the projects?

Folder structure:

```
.git
package.json
js-modules
  modul1
    package.json
  modul2
    package.json
  modul3
    package.json
```

Scenario:

1. open the package.json in root
2. find file and select the package.json in modul1
3. the root switched to js-modules/modul1

How is it possible to only detect .git?

Best regards and thx a lot!

**Comment by prabanjanraja** [comment]
find files will not show you the files that are ignored by git

**Reply by codingluke** [reply]
Hi @prabanjanraja thank you for your input. 

I am aware that find files will not work for files listed in .gitignore. However, in my scenario it is not that the files are not found. The problem is, that when you open a file from a submodule (by finding it when open a file in root) it will change the root to the submoduls root. This has the effect, that it is not possible to find a file inside modul2 from within a file in modul1.

**Reply by codingluke** [reply]
I found actually a workaround. With the https://www.lazyvim.org/extras/util/project extra, it is possible to select the project first with `<leader>fp` and search within this. This works with a few handstrokes more.

Nevertheless, I would be still interessted in a find root only by .git solution.

**Reply by folke** [reply, marked as answer]
@codingluke see my other comment. LazyVIM **never** changes the cwd dir
The project extra however did. I just disabled that in its options.

**Reply by dpetka2001** [reply]
@folke I guess you meant Lazyvim never changes the current working directory and not the root. Because root_dir changes based on the LSP attached on the buffer, no? I just tried having nested `package.json` and the root_dir for each buffer changes based on that. Although `<leader>fF` still works correctly and `<leader>ff` according to this. I guess the problem might lie with `Telescope lsp_references` and stuff when you're inside a nested directory with its own `package.json`, in which case it might be better to configure the server to detect `root_dir` by `.git` first?

**Reply by folke** [reply]
yes, I meant cwd. Never changes the cwd

**Comment by dpetka2001** [comment]
#1307 will be of help to you.

**Comment by folke** [comment]
LazyVim **does not** change your working directory!

It's probably that `project` extra that does that.

If anyone knows how to disable it, please submit a PR to LazyVim.

You're not the first that's blaming LazyVim's root detection for this.

Edit: just disabled that in the project extra

**Comment by codingluke** [comment]
I can confirm that the [commit](https://github.com/LazyVim/LazyVim/commit/7d8b3e8ef96f91f96c8e3a0ba62dfab270debf9f) fixes the issue. After this change `<leader>ff` searches in the new root (which is perfectly fine) but `<leader>fF` searchs from the .git root, which is, as I just learned, the cwd and not the root :exploding_head:
